test/runtests.jl

```diff
@@ -2297,6 +2297,56 @@ else  # run HiGHS tests
 
         end
 
+        @testset "ASHP" begin
+            #Case 1: Boiler produces the required heat instead of ASHP - ASHP is not purchased
```

Comment:
Do we have an equivalent way to "force in space heating ASHP", like we do for GHP, right now? I think the new `ExistingBoiler.retire_in_optimal` would "force in space heating **and DHW** ASHP", but we don't have a way to just force in "DHW ASHP".

Reply:
I guess it would be setting the min size, but in order to do that accurately you'd have to know what the peak heating or coincident peak heating + cooling thermal load.

Reply:
Would the force-in function set set all other heating production variables to zero, i.e., would we allow only ASHP to be the source of a given heating load?  I ask because we'd also probably exclude other possible techs like the CHP from providing heat as well.  (And, if we do this, does that mean no storage as well?  CHP would serve load using storage as a middleman otherwise.)

Reply:
Ok, thanks for the clarification on this @Bill-Becker.  I'm going to also include "retire_in_optimal" for the ExistingChiller as well since the ASHP space heating and cooling may replace that as well (and could also compete with the AbsorptionChiller and GHP for serving cooling load).  I'll also add a test for retiring these in the ASHP test suite.

Reply:
Good idea!

Reply:
Ok, with the most recent push this has been implemented via the new function here: 
https://github.com/NREL/REopt.jl/blob/efc94a29a501b0bf3d283cfba63ab3edc01f5896/src/constraints/thermal_tech_constraints.jl#L81-L86

And the new checks for other cooling and heating techs here (this is edited from the original PR with `retire_in_optimal` for the boiler):
https://github.com/NREL/REopt.jl/blob/efc94a29a501b0bf3d283cfba63ab3edc01f5896/src/core/reopt.jl#L299-L335

I have warnings in place instead of errors if (1) there are more heating techs than the existing boiler, (2) the existing boiler is to be retired, and (3) the new techs can't meet all the heating loads without the existing boiler present.  I just warn and run the problem with the boiler instead of an error, but we can raise an exception if you think that's more appropriate.

Reply:
I think we should raise an exception and not allow running the scenario if the new heating techs can't serve all of the specified heating load(s).